cmake/Modules/Platform/wasi.cmake

```diff
@@ -0,0 +1,3 @@
+set(WASI 1)
+set(UNIX 1)
```

Comment:
WASI isn't "Unix" -- we don't predefine `__unix__` and so forth. For code in libc, it's better to just teach the code how to handle WASI than to pretend to be Unix.

Reply:
This is not for defining `__unix__`, this it to correct the handling of library generation inside of CMake itself (it makes assumptions about library prefixes, naming, etc based on this variable).

Reply:
Ok, makes sense. Would you mind adding this in a comment in the code?

Reply:
Absolutely.